src/main/java/core/basesyntax/data/FormaterImpl.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax.data;
+
+import core.basesyntax.operation.Operation;
+import core.basesyntax.transaction.FruitTransaction;
+import java.util.Arrays;
+import java.util.List;
+
+public class FormaterImpl implements Formater {
+    private static final String DEFAULT_DELIMITER = ",";
+    private static final String OPERATION = "type";
+    private static final String FRUIT = "fruit";
+    private static final String QUANTITY = "quantity";
+    private int operationIndex;
+    private int fruitIndex;
+    private int quantityIndex;
+
+    FormaterImpl(String pattern) {
+        String[] elements = pattern.split(DEFAULT_DELIMITER);
+        List<String> list = Arrays.asList(elements);
+        operationIndex = list.indexOf(OPERATION);
+        fruitIndex = list.indexOf(FRUIT);
+        quantityIndex = list.indexOf(QUANTITY);
+    }
```

Comment:
The constructor for 'FormaterImpl' is package-private, which might not be intended as the class is public. If you want to restrict the access, consider using a different access modifier that reflects your intention more clearly.